app/src/app/[lng]/[inventory]/InventoryResultTab/TopEmissionsWidget.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+  Box, Card,
+  Center,
+  CircularProgress,
+  Heading,
+  HStack,
+  Select,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Text,
+  Th,
+  Thead,
+  Tr
+} from "@chakra-ui/react";
+import { TFunction } from "i18next";
+import { InventoryResponse, TopEmission } from "@/util/types";
+import { convertKgToTonnes } from "@/util/helpers";
+import { api } from "@/services/api";
+import groupBy from "lodash/groupBy";
+import { SegmentedProgress, SegmentedProgressValues } from "@/components/SegmentedProgress";
+
+const TitleAndSelector = ({ t }: { t: Function }) => {
+  return <HStack justifyContent="space-between">
+    <Box>
+      <Heading size="sm" marginTop={10} marginBottom={4}>{t("top-emissions")}</Heading>
+    </Box>
+    <Select width={"15vw"} my={2}
+    >
+      {["by-sub-sector"].map((grouping) => (
+        <option key={grouping} value={grouping}>
+          {t(grouping)}
+        </option>
+      ))}
+    </Select>
+  </HStack>;
+};
+
+
+const EmissionsTable = ({ topEmissions }: { topEmissions: TopEmission[] }) => {
+  return (
+    <TableContainer my={4}>
+      <Table variant="simple">
+        <Thead>
+          <Tr>
+            <Th sx={{ "font": "bold", color: "black" }}>Subsector</Th>
+            <Th sx={{ "font": "bold", color: "black" }}>Total Emissions (CO2eq)</Th>
+            <Th sx={{ "font": "bold", color: "black" }}>% of Emissions</Th>
+          </Tr>
+        </Thead>
+        <Tbody>
+          {(topEmissions || []).map((emission, index) => (
+            <Tr key={index}>
+              <Td>
+                <Text fontSize="xl">{emission.subsectorName}</Text>
+                <Text fontSize="sm" color={"gray"}>{emission.sectorName}</Text>
+              </Td>
+              <Td>{convertKgToTonnes(emission.co2eq)}</Td>
+              <Td>{emission.percentage}%</Td>
+            </Tr>
+          ))}
+        </Tbody>
+      </Table>
+    </TableContainer>
+  );
+};
+
+
+const TopEmissionsWidget = ({
+                              t,
+                              inventory
+                            }: {
+  t: Function & TFunction<"translation", undefined>;
+  inventory?: InventoryResponse;
+}) => {
+
+  const { data: results, isLoading: isTopEmissionsResponseLoading } =
+    api.useGetResultsQuery(inventory!.inventoryId!);
+
+  function getPercentagesForProgress(): SegmentedProgressValues[] {
```

Comment:
Can we move this into a useMemo ? to avoid rerenders

```suggestion
  const getPercentagesForProgress = useMemo((): SegmentedProgressValues[] => {
    if (!results) return [];

    // Group results by sector
    const grouped = groupBy(results.totalEmissions?.bySector || [], (e) => e.sectorName) as Record<
      string,
      { co2eq: bigint; percentage: number }[]
    >;

    // Map grouped results to the expected format
    return Object.entries(grouped).map(([name, [{ co2eq, percentage }]]) => ({
      name,
      value: co2eq,
      percentage,
    }));
  }, [results]);
```